swarm/api/services/spec_manager.py

```
@@ -503,17 +504,32 @@ def list_runs(self, limit: int = 20) -> List[Dict[str, Any]]:
-        for run_dir in sorted(self.runs_root.iterdir(), reverse=True):
-            if not run_dir.is_dir():
-                continue
+        # Sort directory names lexically in descending order (e.g., run_2, run_1)
+        # We avoid iterdir() because it yields Path objects that require stat() internally
+        # on some systems during sorting/is_dir. scandir is faster for just getting names.
+        candidates = []
+        try:
+            with os.scandir(self.runs_root) as it:
+                for entry in it:
+                    if entry.is_dir():
+                        candidates.append(entry.name)
```

Ruff (0.15.20)

[warning] Use `list.extend` to create a transformed list

(PERF401)